Documentation/contributing/docs/docsframework.rst

```diff
@@ -0,0 +1,382 @@
+``Makefile`` contains the following target (here simplified), which regenerates
+a number of documents and then checks that they are all up-to-date:
+
+.. code-block:: makefile
+
+   check: builder-image api-flaggen update-cmdref update-crdlist update-helm-values update-codeowners
+   	./check-cmdref.sh
+   	./check-helmvalues.sh
+   	$(DOCKER_RUN) ./check-examples.sh   # Runs "cilium policy validate" and "yamllint"
+   	./check-codeowners.sh
+   	./check-flaggen.sh
+   	./check-crdlist.sh
+
+Regeneration happens when the different dependency targets for ``check`` are
+run. They are:
+
+- ``api-flaggen``
+
+  - Runs ``go run tools/apiflaggen``
+  - Generates ``Documentation/configuration/api-restrictions-table.rst``
+
+- ``update-cmdref``
+
+  - Runs ``./update-cmdref.sh``
+  - Includes running various binaries with ``--cmdref``
+  - Generates ``Documentation/cmdref/\*``
+
+- ``update-crdlist``
+
+  - ``make -C ../ generate-crd-docs``
+  - Runs ``tools/crdlistgen/main.go``
+  - Parses docs to list CRDs
+  - Generates ``Documentation/crdlist.rst``
+
+- ``update-helm-values``
+
+  - Generates from ``install/kubernetes``
+  - Generates ``Documentation/helm-values.rst``
+
+- ``update-codeowners``
+
+  - ``./update-codeowners.sh``
+  - Synchronizes teams description from ``CODEOWNERS``
+  - Generates ``Documentation/codeowners.rst``
+
+Other auto-generated contents include:
+
+- OpenAPI reference
+
+  - YAML generated from the ``Makefile`` at the root of the repository
+  - Relies on the contents of ``api``, linked as ``Documentation/_api``
+  - Processed and included via a dedicated add-on, from
+    ``Documentation/api.rst``: ``.. openapi:: ../api/v1/openapi.yaml``
+
+- gRPC API reference
+
+  - Markdown generated from the main ``Makefile`` at the root of the repository
+  - Relies on the contents of ``api``, linked as ``Documentation/_api``
+  - Included from ``Documentation/grpcapi.rst``
+
+Build system
+============
+
+Makefile targets
+----------------
+
+Here are the main ``Makefile`` targets related to documentation, to run from
+the root of the Cilium repository, as well as some indications on what they
+call:
+
+- ``make`` -> ``all: ... postcheck`` -> ``make -C Documentation check``:
+  Build Cilium and validate the documentation via the ``postcheck`` target
+- ``make -C Documentation html``:
+  Render the documentation as HTML
+- ``make test-docs`` -> ``make -C Documentation html``:
+  Render the documentation as HTML
+- ``make -C Documentation live-preview``:
+  Build the documentation and start a server for local preview
+- ``make render-docs`` -> ``make -C Documentation live-preview``:
+  Build the documentation and start a server for local preview
+
+Generating documentation
+------------------------
+
+- The ``Makefile`` builds the documentation using the ``docs-builder`` Docker
+  image.
+
+- The build includes running ``check-build.sh``. This script:
+
+  a. Runs the linter (``rstcheck``) by default
```

Comment:
Do we need to specify default behavior here?
```suggestion
  a. Runs the linter (``rstcheck``)
```

Reply:
I can maybe turn this differently, “... unless \`\`SKIP_LINT\`\` is set”